TodoParser.py

```diff
@@ -296,80 +374,72 @@ def _get_file_details(self, file):
-                            curr_issue.start_line = start_line
-                            break
-                        if i != 0 and (hunk_line.group(0).startswith('+') or not hunk_line.group(0).startswith('-')):
-                            start_line += 1
-
-                elif curr_issue:
-                    # Extract other issue information that may exist below the title.
-                    line_labels = self._get_labels(cleaned_line)
-                    line_assignees = self._get_assignees(cleaned_line)
-                    line_milestone = self._get_milestone(cleaned_line)
-                    line_url = self._get_issue_url(cleaned_line)
-                    if line_labels:
-                        curr_issue.labels.extend(line_labels)
-                    elif line_assignees:
-                        curr_issue.assignees.extend(line_assignees)
-                    elif line_milestone:
-                        curr_issue.milestone = line_milestone
-                    elif line_url:
-                        curr_issue.issue_url = line_url
-                        issue_number_search = self.ISSUE_NUMBER_PATTERN.search(line_url)
-                        if issue_number_search:
-                            curr_issue.issue_number = issue_number_search.group(1)
-                    elif len(cleaned_line) and line_status != LineStatus.DELETED:
-                        if self.should_escape:
-                            curr_issue.body.append(self._escape_markdown(cleaned_line))
-                        else:
-                            curr_issue.body.append(cleaned_line)
-                    if not line.startswith('-'):
-                        curr_issue.num_lines += 1
-                if not line_title:
-                    prev_line_title = False
+        comment_lines = comment_block['comment'].split('\n')
+        for line_number_within_comment_block, line in enumerate(comment_lines):
+            line_status, committed_line = self._get_line_status(line)
+            line_statuses.append(line_status)
+            cleaned_line = self._clean_line(committed_line, marker)
+            line_title, ref, identifier, identifier_actual = self._get_title(cleaned_line)
+            if line_title:
+                if prev_line_title and line_status == line_statuses[-2]:
+                    # This means that there is a separate one-line TODO directly above this one.
+                    # We need to store the previous one.
+                    curr_issue.status = line_status
+                    found_issues.append(curr_issue)
+                curr_issue = Issue(
+                    title=line_title,
+                    labels=[],
+                    assignees=[],
+                    milestone=None,
+                    body=[],
+                    hunk=hunk_info['hunk'],
+                    file_name=hunk_info['file'],
+                    start_line=hunk_info['start_line'] + comment_block['start'] + line_number_within_comment_block,
```

Comment:
`start_line` is now a function of the hunk's position within the diff _and_ the comment's position within the hunk. When applicable, it also factors in the comment's position within its comment block.